cpp/include/cudf/wrappers/bool.hpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <utilities/cudf_utils.h>
+
+#include <cub/util_type.cuh>
+
+#include <cstdint>
+#include <iosfwd>
+
+/**---------------------------------------------------------------------------*
+ * @file bools.hpp
+ * @brief Concrete type definition for uint8_t values !0 and 0 representing
+ * boolean true and false respectively.
+ *---------------------------------------------------------------------------**/
+namespace cudf {
+namespace experimental {
+
+struct bool8 {
+
+  using value_type = uint8_t;
+
+  // defaulted bool8 move/copy constructors
+
+  bool8() = default;
+  ~bool8() = default;
+  bool8(bool8 &&) = default;
+  bool8(bool8 const& w) = default;
+  bool8& operator=(bool8&&) = default;
+  bool8& operator=(const bool8&) = default;
+
+  // bool8 constructor that takes non-bool8 types
+
+  template <typename from_type>
+  CUDA_HOST_DEVICE_CALLABLE constexpr explicit bool8(from_type v)
+    : value{static_cast<uint8_t>(static_cast<bool>(v))} {}
+
+  // move/copy assignment operators for non-bool8 types
+
+  template <typename from_type>
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator=(from_type&& rhs) {
+    this->value = static_cast<uint8_t>(static_cast<bool>(std::move(rhs)));
```

Comment:
Let's replace instances of `uint8_t` with `value_type`.
```suggestion
    this->value = static_cast<value_type>(static_cast<bool>(std::move(rhs)));
```

Reply:
Updated :+1: